### PR TITLE
Use Results in Hole2

### LINKED ISSUE #3269: update hole2 analysis to use .results
Follow the new AnalysisBase API and store results in HoleAnalysis in `.results` instead of class attributes.

This should be done for 2.0.0 #3254 so that we don't have to start this new analysis module with deprecations in place.

See omnibus issue #3261 for background and details. See PR #3263 as an example (DensityAnalysis) but note that _no deprecations are necessary_ — just change where the results are stored and update the docs.

- [ ] use attributes in`self.results` instead of `self`
- [ ] update docs
- [ ] add an entry in CHANGELOG under Changes (for completeness)

### PR DESCRIPTION
**Pull request by lilyminium**
Fixes #3269 

Changes made in this Pull Request:
 - Uses Results
 - Tidies some code (e.g. self.frames is now part of AnalysisBase)

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/analysis/hole2/hole.py [hunk 1]
```diff
@@ -513,14 +517,49 @@ class HoleAnalysis(AnalysisBase):
...
+        Alias of :attr:`results.profiles`
+
+        .. deprecated:: 2.0.0
+            This will be removed in MDAnalysis 3.0.0. Please use
+            :attr:`results.profiles` instead.
 
     .. versionadded:: 1.0
 
+
+
```
Comment (PicoCentauri) [inline package/MDAnalysis/analysis/hole2/hole.py:562]:
```python
    .. versionchanged:: 2.0.0
        :attr:`sphpdbs`, :attr:`outfiles` and :attr:`profiles `
        are now stored in a :class:`MDAnalysis.analysis.base.Results` 
        instance.
```

**Review by PicoCentauri — CHANGES_REQUESTED** [review]
Just one minor doc addition.

File: package/MDAnalysis/analysis/hole2/hole.py (hunk 1 above)
Comment (orbeckst) [reply package/MDAnalysis/analysis/hole2/hole.py:562]: please add the versionchanged

File: package/CHANGELOG [hunk 2]
```diff
@@ -175,6 +175,8 @@ Enhancements
     checking if it can be used in parallel analysis. (Issue #2996, PR #2950)
 
 Changes
+  * `analysis.hole2.HoleAnalysis` now stores ``sphpdbs``, ``outfiles``
```
Comment (orbeckst) [inline package/CHANGELOG:178]: also needs an entry in Deprecations

**Review by orbeckst — CHANGES_REQUESTED** [review]
only docs (what @PicoCentauri noted) and CHANGELOG

**Review by orbeckst — APPROVED** [review]

**Comment by orbeckst** [comment]
Waiting for CI... for CHANGELOG :-p.

(And depending on which other of the results PRs goes in, we're going to do it again... unless this one is first.)

**Pull request merged**